### ISSUE TITLE
[BUG] JindoMaster mount point did not recovered from pod migration. 

### ISSUE THREAD

**Issue by SimonCqk** [fluid-cloudnative/fluid#3368]
**What is your environment(Kubernetes version, Fluid version, etc.)**

kubernetes 1.22

**Describe the bug**

dataset controller executes `jindo mount /xx oss://xxx` command into jindomatser when DataSet object becomes ready for the first time, after that, `jindo mount` will never be triggered again, event jindo master pod was migrated/preempted from one node to another.

**What you expect to happen:**

Re-execute `jindo mount` command after jindo master recovered from migration or recreate.

**How to reproduce it**

1. create DataSet and JindoRuntime, wait for them to be ready.
2. simulate preemption or node healing by directly delete pod(jindo master).
3. wait until jindo master becomes Running again and exec into the container to check mountpoint.

**Additional Information**

**Comment by TrafalgarZZZ** [comment]
Maybe we can refer to #1322 for similar solution.

**Comment by SimonCqk** [comment]
> Maybe we can refer to #1322 for similar solution.

Yes it is, and it seems that master still corrupted when recreated or restarted with JindoRuntime, anyway, I'll open an PR soon to fix this.
